### PR TITLE
Add market depth API call

### PR DESCRIPTION
**Pull request by Randall-Coding**
[Haveno issue #140](https://github.com/haveno-dex/haveno/issues/140)

### CODE REVIEW

File: src/HavenoDaemon.test.ts [hunk 1]
```diff
@@ -119,6 +119,77 @@ test("Can get market prices", async () => {
...
+  expect(sellPrices[0]).toEqual(3000);
+  expect(sellPrices[1]).toEqual(4000);
+  expect(sellDepth[0]).toEqual(BigInt(2));
+  expect(sellDepth[1]).toEqual(BigInt(1));
+
+  //  // test invalid currency
+  // await expect(async () => {await alice.getMarketDepth("INVALID_CURRENCY")})
+  //   .rejects
+  //   .toThrow('Currency not found: INVALID_CURRENCY');
+});
```
Comment (Randall-Coding) [inline src/HavenoDaemon.test.ts:191]: Wasn't able to make this work.  Do you think it is necessary?

Comment (woodser) [reply src/HavenoDaemon.test.ts:191]: Yes it's necessary to throw if an invalid currency is given, similar to the price call:

```
  // test invalid currency
  await expect(async () => {await alice.getPrice("INVALID_CURRENCY")})
    .rejects
    .toThrow('Currency not found: INVALID_CURRENCY');
```

**Comment by woodser** [comment]
Please rebase.

File: src/HavenoDaemon.test.ts (hunk 1 above)
Comment (Randall-Coding) [reply src/HavenoDaemon.test.ts:191]: I was having an issue determining what is a valid currency.  Tried a few methods:

1) I looked at the method previously used in CorePriceService.java:66, and this seemed to not apply to the situation of market depth in my mind.  `var marketPrice = priceFeedService.requestAllPrices().get(currencyCode);` checks if a market price exists for a given code, but not necessarily ruling out a valid currency.   can see a situation where the currency is valid but there happens to be no prices available.  Correct me if I'm wrong here please.

2) I looked into the isFiatCurrency and isCryptoCurrency boolean functions.  And on first glance you'd think you could just say `!isFiatCurrency && !isCryptoCurrency` to check for invalid currency, but isCryptoCurrency returns true if the code is not found in Fiat, so there could never be a false value using that construction.

**Comment by woodser** [comment]
Needs rebased again unfortunately.

**Comment by Randall-Coding** [comment]
Will do, I'm focusing on fixing the haveno core logic first before finalizing this.

**Comment by Randall-Coding** [comment]
I rebased the other week, but maybe it needs another?

**Comment by woodser** [comment]
> I rebased the other week, but maybe it needs another?

Yeah, and actually needs rebased one more time after the recent and rather large changes to master.

**Comment by Randall-Coding** [comment]
> > I rebased the other week, but maybe it needs another?
> 
> Yeah, and actually needs rebased one more time after the recent and rather large changes to master.

Ok will do.

**Pull request merged**